src/modules/previewpane/MarkDownPreviewHandler/HTMLParsingExtension.cs

```diff
@@ -14,24 +14,35 @@
 
 namespace MarkdownPreviewHandler
 {
+    /// <summary>
+    /// Callback if extension blocks external images.
+    /// </summary>
+    public delegate void ImagesBlockedCallBack();
+
     /// <summary>
     /// Markdig Extension to process html nodes in markdown AST.
     /// </summary>
     public class HTMLParsingExtension : IMarkdownExtension
     {
+        /// <summary>
+        /// Callback if extension blocks external images.
+        /// </summary>
+        private readonly ImagesBlockedCallBack imagesBlockedCallBack;
+
         /// <summary>
         /// Initializes a new instance of the <see cref="HTMLParsingExtension"/> class.
         /// </summary>
-        public HTMLParsingExtension(string baseUrl = "")
+        public HTMLParsingExtension(ImagesBlockedCallBack imagesBlockedCallBack, string baseUrl = "")
         {
+            this.imagesBlockedCallBack = imagesBlockedCallBack;
             this.BaseUrl = baseUrl;
         }
-
+
         /// <summary>
         /// Gets or sets path to directory containing markdown file.
         /// </summary>
         public string BaseUrl { get; set; }
```

Comment:
`BaseUrl` is being set from outside. So `set ` method for this property should be public. And since this property is modified from outside it should be readable, hence `get` method should also be public.